docs/src/p4est_interfacing.md

```diff
@@ -0,0 +1,333 @@
+    info_pw = PointerWrapper(info)
+
+    # Load tree from global trees array, one-based indexing
+    tree_pw = load_pointerwrapper_tree(info_pw.p4est, info_pw.treeid[] + 1)
+    # Quadrant numbering offset of this quadrant
+    offset = tree_pw.quadrants_offset[]
+    # Global quad ID
+    quad_id = offset + info_pw.quadid[]
+
+    # Access user_data = `lambda`
+    user_data_pw = PointerWrapper(Int, user_data)
+    # Load controller_value = `lambda[quad_id + 1]`
+    controller_value = user_data_pw[quad_id + 1]
+
+    # Access quadrant's user data `[global quad ID, controller_value]`
+    quad_data_pw = PointerWrapper(Int, info_pw.quad.p.user_data[])
+    # Save controller value to quadrant's user data.
+    quad_data_pw[2] = controller_value
+
+    return nothing
+end
+```
+This function is then converted to a `C` function to match the required syntax of the [`p4est_iter_volume_t`](https://p4est.github.io/api/p4est-latest/p4est__iterate_8h.html)/[`p8est_iter_volume_t`](https://p4est.github.io/api/p4est-latest/p8est__iterate_8h.html#a4b19423fb264c674bd4deaf5a7194758) function:
+```C
+/** The prototype for a function that p4est_iterate will execute at every
+ * quadrant local to the current process.
+ * \param [in] info          information about a quadrant provided to the user
+ * \param [in,out] user_data the user context passed to p4est_iterate()
+ */
+typedef void (*p4est_iter_volume_t) (p4est_iter_volume_info_t * info,
+                                     void *user_data);
+
+/** The prototype for a function that p8est_iterate() will execute at every
+ * quadrant local to the current process.
+ * \param [in] info          information about a quadrant provided to the user
+ * \param [in,out] user_data the user context passed to p8est_iterate()
+ */
+typedef void (*p8est_iter_volume_t) (p8est_iter_volume_info_t * info,
+                                     void *user_data);
+```
+To handle both 2D (which is the canonical `p4est`) and 3D (which is within the `p4est` library referred to as `p8est`) a dispatch on the mesh dimensionality is included:
+```julia
+# 2D
+function cfunction(::typeof(copy_to_quad_iter_volume), ::Val{2})
+    @cfunction(copy_to_quad_iter_volume, 
+               # Note the matching signature to the C function: 
+               Cvoid, # Return type
+               # Parameters
+               (Ptr{p4est_iter_volume_info_t}, Ptr{Cvoid}))
+end
+# 3D
+function cfunction(::typeof(copy_to_quad_iter_volume), ::Val{3})
+    @cfunction(copy_to_quad_iter_volume, 
+               # Note the matching signature to the C function
+               Cvoid, # Return type
+               # Parameters
+               (Ptr{p8est_iter_volume_info_t}, Ptr{Cvoid}))
+end
+```
+
+The correct function is then selected via
+```julia
+iter_volume_c = cfunction(copy_to_quad_iter_volume, Val(ndims(mesh)))
+```
+and by calling `iterate_p4est` the indicator values are finally copied to the quadrants/cells/elements of the mesh:
```

Comment:
```suggestion
and by calling `iterate_p4est` the indicator values are finally copied to the cells (quadrants or octants) of the mesh:
```
Meshes do not have "elements" in Trixi.jl-speak, only cells. The elements are part of the solver. We make that distinction to make it unambiguous about what we are talking